From d62a1f1cfb20e9e418a4c608c0b9791ad9e280f6 Mon Sep 17 00:00:00 2001
From: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Date: Tue, 8 Feb 2022 22:03:03 +0200
Subject: [PATCH 6/6] s32: add PATH variables across Makefiles

Create a hierarchy of path variables in s32 Makefiles starting from the
root common point 'plat/nxp/s32' and descending into the specific
board targets. This eliminates the repetition of paths across Makefiles.

Issue: ALB-8443
Upstream-Status: Pending 

Signed-off-by: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32_common.mk                    | 81 ++++++++++---------
 plat/nxp/s32/s32g/s32g2/s32g2.mk              | 12 +--
 .../s32/s32g/s32g2/s32g274ardb2/platform.mk   |  6 +-
 .../s32/s32g/s32g2/s32g2xxaevb/platform.mk    |  6 +-
 plat/nxp/s32/s32g/s32g3/s32g3.mk              | 14 ++--
 .../s32/s32g/s32g3/s32g399ardb3/platform.mk   |  4 +-
 .../s32/s32g/s32g3/s32g3xxaevb/platform.mk    |  4 +-
 plat/nxp/s32/s32g/s32g_common.mk              | 40 ++++-----
 plat/nxp/s32/s32r/s32r.mk                     | 20 ++---
 9 files changed, 103 insertions(+), 84 deletions(-)

diff --git a/plat/nxp/s32/s32_common.mk b/plat/nxp/s32/s32_common.mk
index 6e476c9d4..9eb2aaaf2 100644
--- a/plat/nxp/s32/s32_common.mk
+++ b/plat/nxp/s32/s32_common.mk
@@ -19,6 +19,9 @@ ERRATA_SPECULATIVE_AT	:= 1
 HEXDUMP ?= xxd
 SED ?= sed
 
+S32_PLAT	:= plat/nxp/s32
+S32_DRIVERS	:= drivers/nxp/s32
+
 ifneq ($(S32_PLAT_SOC),)
 $(eval $(call add_define_val,PLAT_$(S32_PLAT_SOC)))
 endif
@@ -29,7 +32,7 @@ $(eval $(call add_define_val,S32G_EMU,$(S32G_EMU)))
 S32GEN1_DRAM_INLINE_ECC	?= 1
 $(eval $(call add_define_val,S32GEN1_DRAM_INLINE_ECC,$(S32GEN1_DRAM_INLINE_ECC)))
 
-DDR_DRV = drivers/nxp/s32/ddr
+DDR_DRV = ${S32_DRIVERS}/ddr
 
 DDR_DRV_SRCS += \
 	${DDR_DRV}/ddr_density.c \
@@ -46,40 +49,40 @@ PLAT_INCLUDES 	+= \
 			-Idrivers \
 			-Iinclude/common/tbbr \
 			-Iinclude/drivers \
-			-Iinclude/drivers/nxp/s32 \
+			-Iinclude/${S32_DRIVERS} \
 			-Iinclude/lib \
 			-Iinclude/lib/libc \
 			-Iinclude/lib/psci \
 			-Iinclude/plat/arm/common \
 			-Iinclude/plat/arm/soc/common \
 			-Iinclude/plat/common \
-			-Iplat/nxp/s32/include \
+			-I${S32_PLAT}/include \
 
 PLAT_BL_COMMON_SOURCES += \
 			${GICV3_SOURCES} \
 			common/fdt_wrappers.c \
 			drivers/nxp/uart/linflexuart.c \
-			plat/nxp/s32/s32_bl_common.c \
-			plat/nxp/s32/s32_dt.c \
-			plat/nxp/s32/s32_lowlevel_common.S \
-			plat/nxp/s32/s32_sramc.c \
-			plat/nxp/s32/s32_sramc_asm.S \
-			plat/nxp/s32/s32_linflexuart.c \
-			plat/nxp/s32/s32_linflexuart_crash.S \
-			plat/nxp/s32/s32_mc_me.c \
-			plat/nxp/s32/s32_ncore.c \
-			plat/nxp/s32/s32_pinctrl.c \
+			${S32_PLAT}/s32_bl_common.c \
+			${S32_PLAT}/s32_dt.c \
+			${S32_PLAT}/s32_lowlevel_common.S \
+			${S32_PLAT}/s32_sramc.c \
+			${S32_PLAT}/s32_sramc_asm.S \
+			${S32_PLAT}/s32_linflexuart.c \
+			${S32_PLAT}/s32_linflexuart_crash.S \
+			${S32_PLAT}/s32_mc_me.c \
+			${S32_PLAT}/s32_ncore.c \
+			${S32_PLAT}/s32_pinctrl.c \
 			drivers/delay_timer/delay_timer.c \
 			drivers/delay_timer/generic_delay_timer.c \
-			drivers/nxp/s32/memory_pool.c \
-			drivers/nxp/s32/clk/early_clocks.c \
-			drivers/nxp/s32/clk/enable_clk.c \
-			drivers/nxp/s32/clk/get_rate.c \
-			drivers/nxp/s32/clk/plat_clk.c \
-			drivers/nxp/s32/clk/s32gen1_clk.c \
-			drivers/nxp/s32/rst/s32gen1_rst.c \
-			drivers/nxp/s32/clk/set_par_rate.c \
-			drivers/nxp/s32/i2c/s32_i2c.c \
+			${S32_DRIVERS}/memory_pool.c \
+			${S32_DRIVERS}/clk/early_clocks.c \
+			${S32_DRIVERS}/clk/enable_clk.c \
+			${S32_DRIVERS}/clk/get_rate.c \
+			${S32_DRIVERS}/clk/plat_clk.c \
+			${S32_DRIVERS}/clk/s32gen1_clk.c \
+			${S32_DRIVERS}/rst/s32gen1_rst.c \
+			${S32_DRIVERS}/clk/set_par_rate.c \
+			${S32_DRIVERS}/i2c/s32_i2c.c \
 			${BOOT_INFO_SRC} \
 
 BL2_SOURCES += \
@@ -90,13 +93,13 @@ BL2_SOURCES += \
 			drivers/io/io_fip.c \
 			drivers/io/io_storage.c \
 			drivers/mmc/mmc.c \
-			drivers/nxp/s32/io/io_mmc.c \
-			drivers/nxp/s32/io/io_memmap.c \
-			drivers/nxp/s32/mmc/s32_mmc.c \
+			${S32_DRIVERS}/io/io_mmc.c \
+			${S32_DRIVERS}/io/io_memmap.c \
+			${S32_DRIVERS}/mmc/s32_mmc.c \
 			lib/optee/optee_utils.c \
-			plat/nxp/s32/s32_bl2_el3.c \
-			plat/nxp/s32/s32_storage.c \
-			plat/nxp/s32/s32_lowlevel_bl2.S \
+			${S32_PLAT}/s32_bl2_el3.c \
+			${S32_PLAT}/s32_storage.c \
+			${S32_PLAT}/s32_lowlevel_bl2.S \
 
 BL31_SOURCES += \
 			${XLAT_TABLES_LIB_SRCS} \
@@ -104,19 +107,19 @@ BL31_SOURCES += \
 			drivers/scmi-msg/clock.c \
 			drivers/scmi-msg/entry.c \
 			drivers/scmi-msg/reset_domain.c \
-			drivers/nxp/s32/clk/clk.c \
-			drivers/nxp/s32/clk/fixed_clk.c \
-			drivers/nxp/s32/clk/s32gen1_scmi_clk.c \
-			drivers/nxp/s32/clk/s32gen1_scmi_ids.c \
+			${S32_DRIVERS}/clk/clk.c \
+			${S32_DRIVERS}/clk/fixed_clk.c \
+			${S32_DRIVERS}/clk/s32gen1_scmi_clk.c \
+			${S32_DRIVERS}/clk/s32gen1_scmi_ids.c \
 			plat/common/plat_gicv3.c \
 			plat/common/plat_psci_common.c \
-			plat/nxp/s32/include/plat_macros.S \
-			plat/nxp/s32/s32_bl31.c \
-			plat/nxp/s32/s32_lowlevel_bl31.S \
-			plat/nxp/s32/s32_scmi_clk.c \
-			plat/nxp/s32/s32_scmi_rst.c \
-			plat/nxp/s32/s32_svc.c \
-			plat/nxp/s32/s32_psci.c \
+			${S32_PLAT}/include/plat_macros.S \
+			${S32_PLAT}/s32_bl31.c \
+			${S32_PLAT}/s32_lowlevel_bl31.S \
+			${S32_PLAT}/s32_scmi_clk.c \
+			${S32_PLAT}/s32_scmi_rst.c \
+			${S32_PLAT}/s32_svc.c \
+			${S32_PLAT}/s32_psci.c \
 
 DTC_FLAGS		+= -Wno-unit_address_vs_reg
 
diff --git a/plat/nxp/s32/s32g/s32g2/s32g2.mk b/plat/nxp/s32/s32g/s32g2/s32g2.mk
index 701a81f42..6c22d5d43 100644
--- a/plat/nxp/s32/s32g/s32g2/s32g2.mk
+++ b/plat/nxp/s32/s32g/s32g2/s32g2.mk
@@ -14,12 +14,14 @@ DDR_DRV_SRCS            += ${DDR_DRV}/s32g2/ddrc_cfg.c \
 
 include plat/nxp/s32/s32g/s32g_common.mk
 
-PLAT_INCLUDES		+= -Iplat/nxp/s32/s32g/s32g2/include \
+PLAT_SOC_PATH	:= ${S32_SOC_FAMILY}/${S32_PLAT_SOC}
 
-PLAT_BL_COMMON_SOURCES	+= drivers/nxp/s32/clk/s32g274a_clk.c \
-			   plat/nxp/s32/s32gen1_mc_me.c \
-			   plat/nxp/s32/s32gen1_mc_rgm.c \
-			   plat/nxp/s32/s32gen1_sramc.c \
+PLAT_INCLUDES		+= -I${PLAT_SOC_PATH}/include \
+
+PLAT_BL_COMMON_SOURCES	+= ${S32_DRIVERS}/clk/s32g274a_clk.c \
+			   ${S32_PLAT}/s32gen1_mc_me.c \
+			   ${S32_PLAT}/s32gen1_mc_rgm.c \
+			   ${S32_PLAT}/s32gen1_sramc.c \
 			   lib/cpus/aarch64/s32.S \
 			   lib/cpus/aarch64/cortex_a53.S \
 
diff --git a/plat/nxp/s32/s32g/s32g2/s32g274ardb2/platform.mk b/plat/nxp/s32/s32g/s32g2/s32g274ardb2/platform.mk
index 093543c84..da8beca59 100644
--- a/plat/nxp/s32/s32g/s32g2/s32g274ardb2/platform.mk
+++ b/plat/nxp/s32/s32g/s32g2/s32g274ardb2/platform.mk
@@ -6,8 +6,10 @@
 
 include plat/nxp/s32/s32g/s32g2/s32g2.mk
 
-PLAT_INCLUDES		+= -Iplat/nxp/s32/s32g/s32g2/s32g274ardb2/include \
+S32_BOARD_PATH		:= ${PLAT_SOC_PATH}/s32g274ardb2
 
-PLAT_BL_COMMON_SOURCES	+=	plat/nxp/s32/s32g/s32g2/s32g274ardb2/s32g274ardb2_vr5510.c \
+PLAT_INCLUDES		+= -I${S32_BOARD_PATH}/include \
+
+PLAT_BL_COMMON_SOURCES	+=	${S32_BOARD_PATH}/s32g274ardb2_vr5510.c \
 
 DTB_FILE_NAME		?= fsl-s32g274a-rdb2.dtb
\ No newline at end of file
diff --git a/plat/nxp/s32/s32g/s32g2/s32g2xxaevb/platform.mk b/plat/nxp/s32/s32g/s32g2/s32g2xxaevb/platform.mk
index 1257e1d59..06733bd41 100644
--- a/plat/nxp/s32/s32g/s32g2/s32g2xxaevb/platform.mk
+++ b/plat/nxp/s32/s32g/s32g2/s32g2xxaevb/platform.mk
@@ -6,8 +6,10 @@
 
 include plat/nxp/s32/s32g/s32g2/s32g2.mk
 
-PLAT_INCLUDES		+= -Iplat/nxp/s32/s32g/s32g2/s32g2xxaevb/include \
+S32_BOARD_PATH		:= ${PLAT_SOC_PATH}/s32g2xxaevb
 
-PLAT_BL_COMMON_SOURCES	+=	plat/nxp/s32/s32g/s32g2/s32g2xxaevb/s32g2xxaevb_vr5510.c \
+PLAT_INCLUDES		+= -I${S32_BOARD_PATH}/include \
+
+PLAT_BL_COMMON_SOURCES	+= ${S32_BOARD_PATH}/s32g2xxaevb_vr5510.c \
 
 DTB_FILE_NAME		?= fsl-s32g2xxa-evb.dtb
\ No newline at end of file
diff --git a/plat/nxp/s32/s32g/s32g3/s32g3.mk b/plat/nxp/s32/s32g/s32g3/s32g3.mk
index 4980a8445..7c159cf69 100644
--- a/plat/nxp/s32/s32g/s32g3/s32g3.mk
+++ b/plat/nxp/s32/s32g/s32g3/s32g3.mk
@@ -14,11 +14,13 @@ DDR_DRV_SRCS            += ${DDR_DRV}/s32g3/ddrc_cfg.c \
 
 include plat/nxp/s32/s32g/s32g_common.mk
 
-PLAT_INCLUDES		+= -Iplat/nxp/s32/s32g/s32g3/include \
+PLAT_SOC_PATH	:= ${S32_SOC_FAMILY}/${S32_PLAT_SOC}
 
-PLAT_BL_COMMON_SOURCES	+= plat/nxp/s32/s32g/s32g3/s32g3_mc_me.c \
-			   plat/nxp/s32/s32g/s32g3/s32g3_mc_rgm.c \
-			   plat/nxp/s32/s32g/s32g3/s32g3_sramc.c \
-			   plat/nxp/s32/s32g/s32g3/s32g3_vr5510.c \
-			   drivers/nxp/s32/clk/s32g3_clk.c \
+PLAT_INCLUDES		+= -I${PLAT_SOC_PATH}/include \
+
+PLAT_BL_COMMON_SOURCES	+= ${PLAT_SOC_PATH}/s32g3_mc_me.c \
+			   ${PLAT_SOC_PATH}/s32g3_mc_rgm.c \
+			   ${PLAT_SOC_PATH}/s32g3_sramc.c \
+			   ${PLAT_SOC_PATH}/s32g3_vr5510.c \
+			   ${S32_DRIVERS}/clk/s32g3_clk.c \
 			   lib/cpus/aarch64/cortex_a53.S \
diff --git a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk b/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk
index 475be5f30..4737c38d2 100644
--- a/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk
+++ b/plat/nxp/s32/s32g/s32g3/s32g399ardb3/platform.mk
@@ -6,6 +6,8 @@
 
 include plat/nxp/s32/s32g/s32g3/s32g3.mk
 
-PLAT_INCLUDES		+= -Iplat/nxp/s32/s32g/s32g3/s32g399ardb3/include \
+S32_BOARD_PATH		:= ${PLAT_SOC_PATH}/s32g399ardb3
+
+PLAT_INCLUDES		+= -I${S32_BOARD_PATH}/include \
 
 DTB_FILE_NAME		?= fsl-s32g399a-rdb3.dtb
diff --git a/plat/nxp/s32/s32g/s32g3/s32g3xxaevb/platform.mk b/plat/nxp/s32/s32g/s32g3/s32g3xxaevb/platform.mk
index 27fcdc57b..d0b35f421 100644
--- a/plat/nxp/s32/s32g/s32g3/s32g3xxaevb/platform.mk
+++ b/plat/nxp/s32/s32g/s32g3/s32g3xxaevb/platform.mk
@@ -6,6 +6,8 @@
 
 include plat/nxp/s32/s32g/s32g3/s32g3.mk
 
-PLAT_INCLUDES		+= -Iplat/nxp/s32/s32g/s32g3/s32g3xxaevb/include \
+S32_BOARD_PATH		:= ${PLAT_SOC_PATH}/s32g3xxaevb
+
+PLAT_INCLUDES		+= -I${S32_BOARD_PATH}/include \
 
 DTB_FILE_NAME		?= fsl-s32g3xxa-evb.dtb
diff --git a/plat/nxp/s32/s32g/s32g_common.mk b/plat/nxp/s32/s32g/s32g_common.mk
index 6977f3a6a..3732e452a 100644
--- a/plat/nxp/s32/s32g/s32g_common.mk
+++ b/plat/nxp/s32/s32g/s32g_common.mk
@@ -6,6 +6,8 @@
 
 include plat/nxp/s32/s32_common.mk
 
+S32_SOC_FAMILY	:= ${S32_PLAT}/s32g
+
 ifeq ($(S32G_EMU),1)
 DDR_DRV_SRCS := \
 	${DDR_DRV}/emu/ddrss_emu.c \
@@ -15,35 +17,35 @@ DDR_DRV_SRCS := \
 endif
 
 ifeq ($(S32G_EMU),0)
-include plat/nxp/s32/s32g/bl31_sram/bl31_sram.mk
-include plat/nxp/s32/s32g/bl31_ssram/bl31_ssram.mk
+include ${S32_SOC_FAMILY}/bl31_sram/bl31_sram.mk
+include ${S32_SOC_FAMILY}/bl31_ssram/bl31_ssram.mk
 endif
 
-PLAT_INCLUDES		+= -Iplat/nxp/s32/s32g/include \
-			   -Iplat/nxp/s32/s32g/bl31_sram/include \
-			   -Iplat/nxp/s32/s32g/bl31_ssram/include \
+PLAT_INCLUDES		+= -I${S32_SOC_FAMILY}/include \
+			   -I${S32_SOC_FAMILY}/bl31_sram/include \
+			   -I${S32_SOC_FAMILY}/bl31_ssram/include \
 
 PLAT_BL_COMMON_SOURCES	+= \
-			   plat/nxp/s32/s32g/s32g_mc_me.c \
-			   plat/nxp/s32/s32g/s32g_bl_common.c \
-			   plat/nxp/s32/s32g/s32g_pinctrl.c \
-			   plat/nxp/s32/s32g/s32g_clocks.c \
-			   drivers/nxp/s32/clk/s32g_clk.c \
-			   drivers/nxp/s32/ocotp.c \
+			   ${S32_SOC_FAMILY}/s32g_mc_me.c \
+			   ${S32_SOC_FAMILY}/s32g_bl_common.c \
+			   ${S32_SOC_FAMILY}/s32g_pinctrl.c \
+			   ${S32_SOC_FAMILY}/s32g_clocks.c \
+			   ${S32_DRIVERS}/clk/s32g_clk.c \
+			   ${S32_DRIVERS}/ocotp.c \
 			   lib/utils/crc8.c \
-			   plat/nxp/s32/s32g/s32g_vr5510.c \
-			   drivers/nxp/s32/pmic/vr5510.c \
+			   ${S32_SOC_FAMILY}/s32g_vr5510.c \
+			   ${S32_DRIVERS}/pmic/vr5510.c \
 			   ${BL31SRAM_SRC_DUMP} \
 
 BL2_SOURCES		+= \
-			   plat/nxp/s32/s32g/s32g_bl2_el3.c \
+			   ${S32_SOC_FAMILY}/s32g_bl2_el3.c \
 			   ${BL31SSRAM_SRC_DUMP} \
 
-BL31_SOURCES		+= plat/nxp/s32/s32g/s32g_bl31.c \
-			   plat/nxp/s32/s32g/s32g_resume.c \
-			   plat/nxp/s32/s32g/s32g_pm.c \
-			   drivers/nxp/s32/s32g_wkpu.c \
-			   drivers/nxp/s32/clk/s32g_scmi_ids.c \
+BL31_SOURCES		+= ${S32_SOC_FAMILY}/s32g_bl31.c \
+			   ${S32_SOC_FAMILY}/s32g_resume.c \
+			   ${S32_SOC_FAMILY}/s32g_pm.c \
+			   ${S32_DRIVERS}/s32g_wkpu.c \
+			   ${S32_DRIVERS}/clk/s32g_scmi_ids.c \
 
 ### Platform-specific defines ###
 # Which LinFlexD to use as a UART device
diff --git a/plat/nxp/s32/s32r/s32r.mk b/plat/nxp/s32/s32r/s32r.mk
index 01811c910..f5f332273 100644
--- a/plat/nxp/s32/s32r/s32r.mk
+++ b/plat/nxp/s32/s32r/s32r.mk
@@ -14,22 +14,24 @@ DDR_DRV_SRCS            += ${DDR_DRV}/s32r/ddrc_cfg.c \
 
 include plat/nxp/s32/s32_common.mk
 
-PLAT_INCLUDES	+=	-Iplat/nxp/s32/s32r/include \
-					-Iplat/nxp/s32/include \
+PLAT_SOC_PATH	:= ${S32_PLAT}/${S32_PLAT_SOC}
 
-PLAT_BL_COMMON_SOURCES += drivers/nxp/s32/clk/s32r45_clk.c \
-		plat/nxp/s32/s32gen1_mc_me.c \
-		plat/nxp/s32/s32gen1_mc_rgm.c \
-		plat/nxp/s32/s32gen1_sramc.c \
+PLAT_INCLUDES	+=	-I${PLAT_SOC_PATH}/include \
+					-I${S32_PLAT}/include \
+
+PLAT_BL_COMMON_SOURCES += ${S32_DRIVERS}/clk/s32r45_clk.c \
+		${S32_PLAT}/s32gen1_mc_me.c \
+		${S32_PLAT}/s32gen1_mc_rgm.c \
+		${S32_PLAT}/s32gen1_sramc.c \
 		lib/cpus/aarch64/s32.S \
 		lib/cpus/aarch64/cortex_a53.S \
 
 BL2_SOURCES 	+=  \
-	plat/nxp/s32/s32r/s32r_bl2_el3.c \
+	${PLAT_SOC_PATH}/s32r_bl2_el3.c \
 
 BL31_SOURCES += \
-		   drivers/nxp/s32/clk/s32r_scmi_ids.c \
-	       plat/nxp/s32/s32r/s32r_bl31.c \
+		   ${S32_DRIVERS}/clk/s32r_scmi_ids.c \
+	       ${PLAT_SOC_PATH}/s32r_bl31.c \
 
 ERRATA_S32_050481	:= 1
 ERRATA_S32_050543	:= 1
-- 
2.17.1

